src/contract/Contract.ts

```diff
@@ -84,8 +92,9 @@ export interface Contract<State = unknown> extends Source {
    * Connecting a signer MUST be done before "writeInteraction" and "bundleInteraction".
    *
    * @param signer - either {@link ArWallet} that will be connected to this contract or custom {@link SigningFunction}
+   * @param signatureType - type of wallet to connect to - 'arweave' or 'ethereum'
    */
-  connect(signer: ArWallet | SigningFunction): Contract<State>;
+  connect(signature: Signature): Contract<State>;
```

Comment:
tu źle.

Miało być:
`signer: ArWallet | Signature`

gdzie `Signature` to
```
export interface Signature {
  signer: SigningFunction;
  signatureType: 'arweave' | 'ethereum';
}
```

Reply:
... w przypadku przekazania jako `ArWallet`, poza generowaniem funkcji podpisującej, ustawiamy też automatycznie `signatureType` na `arweave`.